### PR TITLE
DRY up JSON rendering

### PR DESCRIPTION
**Pull request by azdagron**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -430,13 +396,66 @@ func showStatsJson(w http.ResponseWriter, r *http.Request) {
...
 	if err != nil {
-		log.Printf("error encoding files: %v", err)
+		what := fmt.Sprintf("unable to render %T: %v", obj, err)
+		log.Println(what)
+		http.Error(w, what, http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+	_, _ = w.Write(jsonData)
```
Comment (stefanbenten) [inline main.go]: Shouldnt we log the error?

Comment (azdagron) [reply main.go]: Sure. I can do that.

**Comment by azdagron** [comment]
Ideally none of the "clean" stuff would not be necessary. I added it because the javascript has a bunch of bugs when slices are returned as `null`, since it does no validation and starts using those fields as arrays.

As written though, even though a bit complicated, it automatically turns nil slices into zero-sized slices so they are rendered as `[]` in the JSON, preventing the bugs in the javascript from being manifest. One side or the other has to do this sort of thing. Go generally treats nil slices the same as an empty slice so it looks weird to do special casing everywhere one of these structs is made to ensure the slice is never nil just for the sake of the javascript.

File: main.go (hunk 1 above)
Comment (azdagron) [reply main.go]: Actually, there are several other spots that don't log writing the HTTP body (I personally go back and forth on whether or not it is useful). Maybe I can do that cleanup in its own PR?

Comment (stefanbenten) [reply main.go]: Yeah, sure. Its not critical by any stretch either!

**Comment by stefanbenten** [comment]
Yeah fair point. I wasnt thinking about those cases.

**Pull request merged**